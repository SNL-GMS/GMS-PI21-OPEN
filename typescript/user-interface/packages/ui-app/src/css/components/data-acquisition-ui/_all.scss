@import 'soh-vars';
@import 'soh-cell';
@import 'ag-theme-overrides.scss';
@import 'data-acquisition';
@import 'environment-history';
@import 'historical-trends-history';
@import 'soh-chart';
@import 'station-statistics';
@import 'soh-lag-missing';
@import 'soh-environment';
@import 'soh-overview';
@import 'soh-quiet-indicator';
@import 'capability-counts';
@import 'soh-map';
