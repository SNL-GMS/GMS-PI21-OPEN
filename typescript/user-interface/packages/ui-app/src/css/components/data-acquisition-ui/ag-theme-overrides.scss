.ag-theme-dark--soh {
  border-radius: $cell-cap-radius;

  .ag-header {
    background: transparent;
  }

  .ag-header-cell {
    .ag-react-container {
      width: 100%;
    }
  }
  .ag-header-viewport {
    background-color: var(--gms-prominent-background, $gms-prominent-background);
  }

  .ag-row {
    overflow: hidden;
    border-radius: $cell-cap-radius;
  }

  .ag-ltr .ag-cell:not(.ag-cell-focus).ag-cell-last-left-pinned,
  .ag-ltr .ag-row.ag-cell-last-left-pinned,
  .ag-horizontal-left-spacer {
    border-right: none;
  }

  &:not(.table--cell-selection-only) {
    .ag-row-hover .ag-cell {
      @include brighten;
    }
  }
  &.table--cell-selection-only {
    .ag-cell:hover {
      @include brighten;
    }
  }

  // TODO: Make this dynamic so it doesn't expect a fixed
  // number of columns
  .ag-row-selected {
    .ag-cell .table-cell {
      box-sizing: border-box;
      @include selectedBorder('top', 'bottom');
    }
    .ag-cell[aria-colindex='1'] .table-cell {
      @include selectedBorder('left');
    }
    .ag-cell .table-cell[data-col-position='last'] {
      @include selectedBorder('right');
    }
  }

  .ag-body-viewport,
  .ag-root-wrapper {
    background: transparent;
  }
}
