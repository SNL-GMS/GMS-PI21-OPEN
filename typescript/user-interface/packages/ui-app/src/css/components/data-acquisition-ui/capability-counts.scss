.stacked-counts {
  @include flexColumn;
  @include centerChildren;
  &__entry {
    padding: 0;
    margin: 0;
    line-height: 1em;
    padding-right: 0.25rem;
    margin-left: -0.25rem;
    font-weight: 400;
    opacity: 0.8;
    font-size: 0.875em;
    & + & {
      margin-top: 0.125rem;
    }
  }
}

// Color the marginal capability count yellow if it is inside of a red cell
[data-cell-status='bad'] .stacked-counts__entry[data-capability-status='marginal'] {
  color: $marginal-color;
}
