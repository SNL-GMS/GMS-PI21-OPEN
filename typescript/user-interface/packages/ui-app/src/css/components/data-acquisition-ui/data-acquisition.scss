.form-field-small {
  width: 60%;
}

.form-field-large {
  margin-left: 5%;
  width: 80%;
}

.footer-button-row {
  position: absolute;
  top: 90%;
  left: 1.5%;
}

.footer-button {
  position: relative;
  display: inline;
}

.footer-button-middle {
  position: relative;
  display: inline;
  left: 200%;
}

.footer-button-right {
  position: relative;
  display: inline;
  left: 225%;
}

.station-configuration-form__row {
  display: flex;
  flex-direction: row;
  margin-bottom: 1%;
}

.station-configuration-form {
  display: flex;
  position: absolute;
  top: 0%;
  height: 20%;
  right: 0%;
  left: 0.5%;
  flex-direction: column;
  width: 100%;
  overflow-y: scroll;
}

.station-configuration-inner-menu {
  position: absolute;
  top: 45%;
  height: 20%;
  right: 0%;
  left: 1.5%;
}

.station-configuration-bottom-menu {
  position: absolute;
  top: 68%;
  height: 20%;
  right: 0%;
  left: 1.5%;
}

.station-configuration-form__labels-top-left {
  margin-left: 1% !important;
  margin-top: 0.5% !important;
  margin-right: 1% !important;
  font-size: small;
}

.station-configuration-form__labels-top-right {
  margin-left: 1% !important;
  margin-top: 1% !important;
  font-size: small;
}

.station-configuration-form__labels-middle {
  margin-left: 1% !important;
  margin-right: 15px !important;
  font-size: small;
}

.station-configuration-form__field-top {
  margin-left: 1.5% !important;
  margin-top: 15px !important;
  font-size: small;
}

.station-configuration-form__field-middle {
  margin-left: 12px !important;
  margin-top: 5px !important;
  font-size: small;
  width: 200%;
}

.station-configuration-form__field-top {
  margin-left: 5px !important;
  margin-top: 5px !important;
  font-size: small;
}

.station-configuration-form__field-bottom {
  margin-left: 3px !important;
  margin-top: 5px !important;
  font-size: small;
}

.station-configuration-form__labels-bottom-right {
  margin-left: 1% !important;
  font-size: small;
}

.station-configuration-form__labels-bottom-left {
  margin-left: 1% !important;
  font-size: small;
}

.station-configuration-form__input-middle {
  padding-left: 10% !important;
  width: 200%;
}

.configure-station-groups-select-network {
  position: relative;
  top: 0%;
  height: 100%;
  width: 40%;
  right: 0%;
  left: 0%;
}

.configure-station-groups-select-network-table {
  position: relative;
  top: 0%;
  height: 100%;
  max-height: calc(100% - 49px);
  width: 100%;
  right: 0%;
  left: 0%;
}

.configure-station-groups-edit-stations {
  position: absolute;
  top: 0%;
  height: 98%;
  right: 0%;
  left: 41%;
  visibility: hidden;
}

.configure-station-groups-no-network-selected {
  position: absolute;
  top: 0%;
  height: 98%;
  right: 0%;
  left: 41%;
  visibility: visible;
}

.configure-station-groups-associated-stations {
  position: absolute;
  top: 0%;
  height: 100%;
  width: 45%;
  right: 0%;
  left: 0%;
}

.configure-station-groups-arrows {
  position: absolute;
  top: 40%;
  height: 100%;
  width: 4%;
  right: 0%;
  left: 48%;
}

.configure-station-groups-available-stations {
  position: absolute;
  top: 0%;
  height: 100%;
  width: 45%;
  right: 0%;
  left: 55%;
}

.data-acquisition-popover {
  font-size: small !important;
  max-height: 300px !important;
  overflow-y: auto !important;
  margin: 0.5rem;
  display: inline-table;
}

.data-acquisition-import-css-popover-with-table {
  font-size: small !important;
  max-height: 500px !important;
  overflow-y: scroll !important;
  margin: 0.5rem;
}

.data-acquisition-inputs input {
  border: 1px solid var(--gms-main, $gms-main);
}

.data-acquisition-inputs input:focus {
  outline: none !important;
  border: 1px solid $data-acq-strong-warning;
}

.data-acquisition-cell .ag-cell {
  display: flex;
  justify-content: center;
  align-items: center !important;
}

.soh-update-time-controls {
  flex: 0 0 auto;
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  color: var(--gms-main, $gms-main) !important;
}

.soh-update-time-controls__label {
  margin-bottom: 0rem !important;
  margin-left: 1rem !important;
  color: var(--gms-main, $gms-main);
}

.soh-update-time {
  font-family: $gms-mono;
  color: var(--gms-main, $gms-main);
}

.text-opacity-80 {
  opacity: 0.8;
}

.soh-drill-down-station-label {
  font-size: 1.5em;
  margin-top: 0.25em;
  padding-left: 0.125em;
  text-rendering: geometricPrecision;
}

.display-title {
  padding: 0.25em 0 0 0.25em;
  display: flex;
  align-items: baseline;
  &__subtitle {
    margin-left: 0.5em;
    color: var(--gms-prominent, $gms-prominent);
    font-size: 1.1rem;
    font-weight: 100;
    @include ellipsis;
  }
}

.soh-table-label {
  padding: 0.5em 0 0.25em 0;
  color: var(--gms-prominent, $gms-prominent);
  margin-left: 0.25em;
  font-size: 1.1rem;
  font-weight: 400;
  font-family: $gms-sans;
  -webkit-font-smoothing: antialiased;
}

.drop-zone__wrapper .drop-zone {
  &--highlighted {
    background-color: var(--gms-table-selection, $gms-table-selection);
  }
}
