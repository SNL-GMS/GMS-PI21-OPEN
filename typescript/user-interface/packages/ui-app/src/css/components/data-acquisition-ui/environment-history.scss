.environment-history-display {
  &__chart-container {
    height: auto;
    width: 100%;
    overflow-x: auto;
  }
  .external-axis {
    min-height: 50px;
    background-color: var(--gms-prominent-background, $gms-prominent-background);
    border-radius: $rounded-rectangle-radius;
    overflow-y: hidden;
    overflow-x: auto;
  }
}
