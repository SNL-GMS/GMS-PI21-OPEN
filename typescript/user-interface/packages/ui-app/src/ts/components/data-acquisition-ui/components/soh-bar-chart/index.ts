export { SohLag } from './soh-bar-chart-container';
export { SohMissing } from './soh-bar-chart-container';
export { SohTimeliness } from './soh-bar-chart-container';
