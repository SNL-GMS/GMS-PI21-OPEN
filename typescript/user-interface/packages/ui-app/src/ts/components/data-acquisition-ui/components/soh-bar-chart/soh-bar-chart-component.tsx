/* eslint-disable react/destructuring-assignment */
import type { SohTypes } from '@gms/common-model';
import type { ValueType } from '@gms/common-util';
import type GoldenLayout from '@gms/golden-layout';
import type { AppState } from '@gms/ui-state';
import { setSelectedStationIds, ssamControlApiSlice } from '@gms/ui-state';
import * as React from 'react';
import type { ConnectedProps } from 'react-redux';
import { connect } from 'react-redux';

import { BaseDisplay } from '~components/common-ui/components/base-display';
import { dataAcquisitionUIConfig } from '~components/data-acquisition-ui/config';

import type { Type } from './bar-chart/types';
import { SohBarChartPanel } from './soh-bar-chart-panel';

const MIN_CHART_HEIGHT_PX = dataAcquisitionUIConfig.dataAcquisitionUserPreferences.minChartHeightPx;

/**
 * Mapping redux state to the properties of the component
 *
 * @param state App state, root level redux store
 */
const mapStateToProps = (type: Type, valueType: ValueType) => (state: AppState) => ({
  type,
  valueType,
  selectedStationIds: state.app.common.selectedStationIds,
  sohStatus: state.app.dataAcquisition.data.sohStatus,
  sohConfigurationQuery: ssamControlApiSlice.endpoints.getSohConfiguration.select()(state)
});

/**
 * Mapping methods (actions and operations) to dispatch one or more updates to the redux store
 *
 * @param dispatch the redux dispatch event alerting the store has changed
 */
const mapDispatchToProps = {
  setSelectedStationIds,
  getSohConfiguration: ssamControlApiSlice.endpoints.getSohConfiguration.initiate
};

export const sohBarChartConnector = (type: Type, valueType: ValueType) =>
  connect(mapStateToProps(type, valueType), mapDispatchToProps);

type ConnectedReduxProps = ConnectedProps<ReturnType<typeof sohBarChartConnector>>;

/**
 * SohBarChartProps props
 */
export type SohBarChartProps = ConnectedReduxProps & {
  glContainer?: GoldenLayout.Container;
  type: Type;
  valueType: ValueType;
};

export class SohBarChart extends React.Component<SohBarChartProps> {
  // ***************************************
  // BEGIN REACT COMPONENT LIFECYCLE METHODS
  // ***************************************
  private unsubscribe: null | (() => void) = null;

  public componentDidMount(): void {
    const { getSohConfiguration } = this.props;
    // eslint-disable-next-line @typescript-eslint/unbound-method
    const { unsubscribe } = getSohConfiguration();
    this.unsubscribe = unsubscribe;
  }

  public componentWillUnmount(): void {
    this.unsubscribe?.();
  }

  /**
   * Renders the component.
   */
  // eslint-disable-next-line react/sort-comp
  public render(): JSX.Element {
    return (
      <BaseDisplay
        glContainer={this.props.glContainer}
        className={`${this.props.type.toLocaleLowerCase()}-display top-level-container scroll-box scroll-box--y`}
      >
        <SohBarChartPanel
          minHeightPx={MIN_CHART_HEIGHT_PX}
          glContainer={this.props.glContainer}
          type={this.props.type}
          station={this.getStation()}
          sohStatus={this.props.sohStatus}
          sohConfiguration={this.props.sohConfigurationQuery.data}
          valueType={this.props.valueType}
        />
      </BaseDisplay>
    );
  }
  // ***************************************
  // END REACT COMPONENT LIFECYCLE METHODS
  // ***************************************

  /** Returns the selected station */
  private readonly getStation = (): SohTypes.UiStationSoh =>
    this.props.sohStatus.stationAndStationGroupSoh.stationSoh.find(
      s => s.stationName === this.props.selectedStationIds[0]
    );
}
