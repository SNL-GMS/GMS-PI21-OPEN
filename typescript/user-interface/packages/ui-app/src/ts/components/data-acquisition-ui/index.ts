import * as DataAcquisitionComponents from './components';

export { DataAcquisitionComponents };
