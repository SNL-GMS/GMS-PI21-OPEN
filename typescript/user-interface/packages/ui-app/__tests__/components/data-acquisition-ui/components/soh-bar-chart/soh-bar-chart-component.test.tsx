import { uuid } from '@gms/common-util';
import type { Container } from '@gms/golden-layout';
import { dataAcquisitionActions, getStore } from '@gms/ui-state';
import { sohConfiguration } from '@gms/ui-state/__tests__/__data__/soh-configuration-query-data';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';

import type { SohBarChartProps } from '../../../../../src/ts/components/data-acquisition-ui/components/soh-bar-chart/soh-bar-chart-component';
import { SohBarChart } from '../../../../../src/ts/components/data-acquisition-ui/components/soh-bar-chart/soh-bar-chart-component';
import { channelSoh } from '../../../../__data__/data-acquisition-ui/soh-overview-data';
import { useQueryStateResult } from '../../../../__data__/test-util-data';

// mock the uuid
uuid.asString = jest.fn().mockImplementation(() => '12345789');

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

jest.mock('@gms/ui-state', () => {
  const actual = jest.requireActual('@gms/ui-state');
  return {
    ...actual,
    useGetSohConfigurationQuery: jest.fn(() => ({
      data: sohConfiguration,
      isLoading: false
    }))
  };
});

uuid.asString = jest.fn().mockReturnValue('1e872474-b19f-4325-9350-e217a6feddc0');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();
window.ResizeObserver = jest.fn(() => {
  return { observe: jest.fn(), disconnect: jest.fn(), unobserve: jest.fn() };
});
const globalAny: any = global;
globalAny.ResizeObserver = window.ResizeObserver;
globalAny.DOMRect = jest.fn(() => ({}));

describe('SohBarChart class', () => {
  // eslint-disable-next-line @typescript-eslint/no-magic-numbers
  global.Date.now = jest.fn(() => 1530518207007);

  const glWidth = 1000;
  const glHeight = 500;

  const selectedStationIds = ['A'];

  const setSelectedStationIds = jest.fn();

  const myGLContainer: Container = {
    // Container
    width: glWidth,
    height: glHeight,
    parent: undefined,
    tab: undefined,
    title: 'container-title',
    layoutManager: undefined,
    isHidden: false,
    setState: jest.fn(),
    extendState: jest.fn(),
    getState: jest.fn(),
    getElement: jest.fn(),
    hide: jest.fn(),
    show: jest.fn(),
    setSize: jest.fn(),
    setTitle: jest.fn(),
    close: jest.fn(),
    // EventEmitter
    on: jest.fn(),
    emit: jest.fn(),
    trigger: jest.fn(),
    unbind: jest.fn(),
    off: jest.fn()
  };
  const myEnvReduxProps: any = {
    glContainer: myGLContainer,
    selectedStationIds,
    setSelectedStationIds
  };

  const sohStationAndGroupStatusQuery: any = {
    loading: false,
    stationAndStationGroupSoh: {
      stationSoh: [
        {
          stationName: 'A'
        }
      ]
    }
  };

  const channelSohForStationQuery: any = {
    channelSohForStation: {
      channelSohs: [channelSoh],
      stationName: 'A'
    }
  };

  const sohStatus: any = {
    loading: false,
    stationAndStationGroupSoh: {
      stationSoh: [
        {
          channelSohs: [channelSoh],
          stationName: 'A'
        }
      ]
    }
  };

  const sohConfigurationQuery = useQueryStateResult;
  sohConfigurationQuery.data = sohConfiguration;

  const myProps: SohBarChartProps = {
    sohStatus,
    ...myEnvReduxProps,
    channelSohForStationQuery,
    sohStationAndGroupStatusQuery,
    sohConfigurationQuery,
    saveStationGroupSohStatus: jest.fn(),
    quietChannelMonitorStatuses: jest.fn(async () => new Promise(jest.fn())),
    mutate: undefined,
    result: undefined,
    getSohConfiguration: jest.fn(() => ({ unsubscribe: jest.fn() })) as any
  };

  const store = getStore();
  store.dispatch(
    dataAcquisitionActions.setSohStatus({
      isStale: false,
      lastUpdated: 0,
      loading: false,
      stationAndStationGroupSoh: undefined
    })
  );

  const sohBarChart = Enzyme.mount(
    <Provider store={store}>
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <SohBarChart {...{ ...myProps, type: 'LAG' as any }} />
    </Provider>
  );

  it('should be defined', () => {
    expect(SohBarChart).toBeDefined();
  });

  it('should get station info', () => {
    sohBarChart.update();
    const stationLagInfo = sohBarChart.find(SohBarChart).instance().getStation();
    expect(stationLagInfo).toBeDefined();
  });

  // it.skip('should acknowledge channel monitor status', () => {
  //   const stationName = 'AAA';
  //   const sohMonType: SohTypes.SohMonitorType = SohTypes.SohMonitorType.LAG;
  //   const chanMonPair: SohTypes.ChannelMonitorPair = {
  //     channelName: 'AAA111',
  //     monitorType: sohMonType
  //   };
  //   const channelPairs: SohTypes.ChannelMonitorPair[] = [chanMonPair];
  //   // sohBarChart.find(SohBarChart).instance().quietChannelMonitorStatuses(stationName, channelPairs);
  // });

  it('should match snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <SohBarChart {...{ ...myProps, type: 'LAG' as any }} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
