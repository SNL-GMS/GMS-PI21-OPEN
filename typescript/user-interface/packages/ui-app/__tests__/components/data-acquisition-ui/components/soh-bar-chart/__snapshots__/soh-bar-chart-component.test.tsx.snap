// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SohBarChart class should match snapshot 1`] = `
<div>
  <div
    class="base-display lag-display top-level-container scroll-box scroll-box--y"
  >
    <div
      class="missing-wrapper"
      style="min-height: NaNpx;"
    >
      <div
        class="soh-drill-down__header"
      >
        <div
          class="toolbar"
          style="width: 960px;"
        >
          <div
            class="toolbar__left-group"
          >
            <div
              class="toolbar-item toolbar-item__left"
            >
              <div>
                <div
                  data-cy="filter-soh"
                >
                  <button
                    class="bp4-button bp4-align-left toolbar-button"
                    style="width: 200px;"
                    title="Filter by Status"
                    type="button"
                  >
                    <span
                      class="bp4-button-text"
                    >
                      <span>
                        Filter by Status
                      </span>
                      <span
                        aria-hidden="true"
                        class="bp4-icon bp4-icon-chevron-down"
                        icon="chevron-down"
                      >
                        <svg
                          data-icon="chevron-down"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <path
                            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="toolbar__center-group"
          >
            <div
              class="toolbar__whitespace"
              style="width: 944px;"
            />
          </div>
          <div
            class="toolbar__right-group"
          >
            <div
              class="toolbar-item
                    "
            >
              <div>
                <div
                  class="dropdown-container"
                >
                  <span
                    class="dropdown-selector"
                    data-cy="sort-channels-dropdown"
                  >
                    <div
                      class="bp4-html-select"
                    >
                      <select
                        style="min-width: 220px; width: 220px;"
                        title="Sort"
                      >
                        <option
                          value="Lag: highest to lowest"
                        >
                          Lag: highest to lowest
                        </option>
                        <option
                          value="Lag: lowest to highest"
                        >
                          Lag: lowest to highest
                        </option>
                        <option
                          value="Channel: A - Z"
                        >
                          Channel: A - Z
                        </option>
                        <option
                          value="Channel: Z - A"
                        >
                          Channel: Z - A
                        </option>
                      </select>
                      <span
                        class="bp4-icon bp4-icon-double-caret-vertical"
                        icon="double-caret-vertical"
                      >
                        <svg
                          data-icon="double-caret-vertical"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <desc>
                            Open dropdown
                          </desc>
                          <path
                            d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="toolbar-item
                    "
            >
              <div
                class="toolbar-label-value"
                style="margin-left: 1em;"
              >
                <div
                  class="label-value-container"
                >
                  <div
                    class="label-value__label"
                    data-cy="last-updated-label"
                  >
                    Last Updated: 
                  </div>
                  <div
                    class="label-value__value"
                    data-cy="last-updated-value"
                    title="Most recent SOH data received"
                  >
                    1970-01-01 00:00:00
                  </div>
                </div>
              </div>
            </div>
            <div
              class="toolbar-item
                    "
            >
              <div
                class="toolbar-label-value"
              >
                <div
                  class="label-value-container"
                >
                  <div
                    class="label-value__label"
                    data-cy="update-interval-label"
                  >
                    Update Interval: 
                  </div>
                  <div
                    class="label-value__value"
                    data-cy="update-interval-value"
                    title="Interval at which SOH data is processed"
                  >
                    20 seconds
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="soh-drill-down-station-label display-title"
          data-cy="drill-down-display-title"
        >
          A
          <div
            class="display-title__subtitle"
          >
            Current lag per channel
          </div>
        </div>
      </div>
      <div
        class="legend-and-charts"
        data-cy="bar-chart"
      >
        <div
          class="core-chart table-display"
        >
          <div
            class="VictoryContainer"
            style="width: 100%; user-select: none; pointer-events: none; position: relative;"
          >
            <svg
              height="NaN"
              role="img"
              style="pointer-events: all; width: 100%; height: 100%;"
              viewBox="0 0 450 NaN"
              width="450"
            >
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: none; stroke-width: 1;"
                  vector-effect="non-scaling-stroke"
                  x1="63"
                  x2="421"
                  y1="NaN"
                  y2="NaN"
                />
                <g
                  role="presentation"
                >
                  <text
                    direction="inherit"
                    dx="0"
                    id="chart-axis-0-tickLabels-0"
                    x="242"
                    y="NaN"
                  >
                    <tspan
                      dx="0"
                      dy="0"
                      style="fill: none; font-family: var(--core-sans); font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent;"
                      text-anchor="middle"
                      x="242"
                    >
                      AAA111
                    </tspan>
                  </text>
                </g>
              </g>
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: var(--core-soft); stroke-width: 1;"
                  vector-effect="non-scaling-stroke"
                  x1="63"
                  x2="421"
                  y1="NaN"
                  y2="NaN"
                />
                <text
                  direction="inherit"
                  dx="0"
                  id="chart-undefined-1-axisLabel-0"
                  x="242"
                  y="NaN"
                >
                  <tspan
                    dx="0"
                    dy="0"
                    style="font-family: var(--core-sans); font-size: 14px; letter-spacing: normal; padding: 82px; fill: var(--core-prominent); stroke: transparent;"
                    text-anchor="middle"
                    x="242"
                  >
                    Channel Name
                  </tspan>
                </text>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke: var(--core-main); stroke-width: 1; size: 4px;"
                    vector-effect="non-scaling-stroke"
                    x1="242"
                    x2="242"
                    y1="NaN"
                    y2="NaN"
                  />
                  <text
                    direction="inherit"
                    dx="-25"
                    id="chart-undefined-1-tickLabels-0"
                    transform="rotate(-65,242,NaN)"
                    x="242"
                    y="NaN"
                  >
                    <title>
                      1.00
                    </title>
                    <tspan
                      dx="-25"
                      dy="0"
                      style="font-size: 10px; font-family: var(--core-sans); letter-spacing: normal; padding: 10px; fill: var(--core-main); stroke: transparent;"
                      text-anchor="middle"
                      x="242"
                    >
                      AAA111
                    </tspan>
                  </text>
                </g>
              </g>
              <g
                role="presentation"
              >
                <line
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: var(--core-soft); stroke-width: 1;"
                  vector-effect="non-scaling-stroke"
                  x1="63"
                  x2="63"
                  y1="16"
                  y2="NaN"
                />
                <text
                  direction="inherit"
                  dx="0"
                  id="chart-undefined-1-axisLabel-0"
                  transform="rotate(-90,23,NaN)"
                  x="23"
                  y="NaN"
                >
                  <tspan
                    dx="0"
                    dy="0"
                    style="font-family: var(--core-sans); font-size: 14px; letter-spacing: normal; padding: 40px; fill: var(--core-prominent); stroke: transparent;"
                    text-anchor="middle"
                    x="23"
                  >
                    Lag (s)
                  </tspan>
                </text>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke: var(--core-main); stroke-width: 1; size: 4px;"
                    vector-effect="non-scaling-stroke"
                    x1="63"
                    x2="59"
                    y1="NaN"
                    y2="NaN"
                  />
                  <text
                    direction="inherit"
                    dx="6"
                    id="chart-undefined-1-tickLabels-0"
                    x="49"
                    y="NaN"
                  >
                    <title>
                      2
                    </title>
                    <tspan
                      dx="6"
                      dy="0"
                      style="font-size: 10px; font-family: var(--core-sans); letter-spacing: normal; padding: 10px; fill: var(--core-main); stroke: transparent;"
                      text-anchor="end"
                      x="49"
                    >
                      2
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke: var(--core-main); stroke-width: 1; size: 4px;"
                    vector-effect="non-scaling-stroke"
                    x1="63"
                    x2="59"
                    y1="NaN"
                    y2="NaN"
                  />
                  <text
                    direction="inherit"
                    dx="6"
                    id="chart-undefined-1-tickLabels-1"
                    x="49"
                    y="NaN"
                  >
                    <title>
                      4
                    </title>
                    <tspan
                      dx="6"
                      dy="0"
                      style="font-size: 10px; font-family: var(--core-sans); letter-spacing: normal; padding: 10px; fill: var(--core-main); stroke: transparent;"
                      text-anchor="end"
                      x="49"
                    >
                      4
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke: var(--core-main); stroke-width: 1; size: 4px;"
                    vector-effect="non-scaling-stroke"
                    x1="63"
                    x2="59"
                    y1="NaN"
                    y2="NaN"
                  />
                  <text
                    direction="inherit"
                    dx="6"
                    id="chart-undefined-1-tickLabels-2"
                    x="49"
                    y="NaN"
                  >
                    <title>
                      6
                    </title>
                    <tspan
                      dx="6"
                      dy="0"
                      style="font-size: 10px; font-family: var(--core-sans); letter-spacing: normal; padding: 10px; fill: var(--core-main); stroke: transparent;"
                      text-anchor="end"
                      x="49"
                    >
                      6
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke: var(--core-main); stroke-width: 1; size: 4px;"
                    vector-effect="non-scaling-stroke"
                    x1="63"
                    x2="59"
                    y1="NaN"
                    y2="NaN"
                  />
                  <text
                    direction="inherit"
                    dx="6"
                    id="chart-undefined-1-tickLabels-3"
                    x="49"
                    y="NaN"
                  >
                    <title>
                      8
                    </title>
                    <tspan
                      dx="6"
                      dy="0"
                      style="font-size: 10px; font-family: var(--core-sans); letter-spacing: normal; padding: 10px; fill: var(--core-main); stroke: transparent;"
                      text-anchor="end"
                      x="49"
                    >
                      8
                    </tspan>
                  </text>
                </g>
                <g
                  role="presentation"
                >
                  <line
                    role="presentation"
                    shape-rendering="auto"
                    style="stroke: var(--core-main); stroke-width: 1; size: 4px;"
                    vector-effect="non-scaling-stroke"
                    x1="63"
                    x2="59"
                    y1="NaN"
                    y2="NaN"
                  />
                  <text
                    direction="inherit"
                    dx="6"
                    id="chart-undefined-1-tickLabels-4"
                    x="49"
                    y="NaN"
                  >
                    <title>
                      10
                    </title>
                    <tspan
                      dx="6"
                      dy="0"
                      style="font-size: 10px; font-family: var(--core-sans); letter-spacing: normal; padding: 10px; fill: var(--core-main); stroke: transparent;"
                      text-anchor="end"
                      x="49"
                    >
                      10
                    </tspan>
                  </text>
                </g>
              </g>
              <g
                role="presentation"
              >
                <path
                  d="M 240, NaN 
A 0 0 0 0 0, 240, NaN 
L 240, NaN 
A 0 0 0 0 0, 240, NaN 
L 244, NaN 
A 0 0 0 0 0, 244, NaN 
L 244, NaN 
A 0 0 0 0 0, 244, NaN 
 z"
                  index="0"
                  role="presentation"
                  shape-rendering="auto"
                  style="stroke: black; padding: 8px; stroke-width: 0;"
                />
                <g
                  transform="translate(137, NaN)"
                >
                  <foreignobject
                    height="60"
                    width="60"
                    x="75"
                    y="0"
                  />
                </g>
              </g>
              <g
                style="width: 100%; user-select: none;"
              >
                <path
                  d="M-894999925705.3741,NaNL894999926189.3741,NaN"
                  role="presentation"
                  shape-rendering="auto"
                  style="fill: none; stroke: #be8c0b; stroke-width: 2;"
                />
              </g>
              <g
                style="width: 100%; user-select: none;"
              >
                <path
                  d="M-894999925705.3741,NaNL894999926189.3741,NaN"
                  role="presentation"
                  shape-rendering="auto"
                  style="fill: none; stroke: #d24c4c; stroke-width: 2;"
                />
              </g>
            </svg>
            <div
              style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
            >
              <svg
                height="NaN"
                style="overflow: visible; width: 100%; height: 100%;"
                viewBox="0 0 450 NaN"
                width="450"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
